src/tests/curve.rs

```diff
@@ -1,383 +1,537 @@
-            assert_eq!(affine_point, affine_point_rec);
-        }
-        {
-            let affine_json = serde_json::to_string(&affine_point).unwrap();
-            let reader = std::io::Cursor::new(affine_json);
-            let affine_point_rec: G::AffineExt = serde_json::from_reader(reader).unwrap();
-            assert_eq!(affine_point, affine_point_rec);
-        }
-        {
-            let projective_bytes = bincode::serialize(&projective_point).unwrap();
-            let reader = std::io::Cursor::new(projective_bytes);
-            let projective_point_rec: G = bincode::deserialize_from(reader).unwrap();
-            assert_eq!(projective_point, projective_point_rec);
-        }
-        {
-            let projective_json = serde_json::to_string(&projective_point).unwrap();
-            let reader = std::io::Cursor::new(projective_json);
-            let projective_point_rec: G = serde_json::from_reader(reader).unwrap();
-            assert_eq!(projective_point, projective_point_rec);
+                assert!(a == a);
+                assert!(b == b);
+                assert!(a != b);
+                assert!(b != a);
+
+                for _ in 0..100 {
+                    let a = $c::random(OsRng);
+                    let b = $c::random(OsRng);
+
+                    assert!(a == a);
+                    assert!(b == b);
+                    assert!(a != b);
+                    assert!(b != a);
+
+                    let a: <$c as CurveExt>::AffineExt = a.into();
+                    let b: <$c as CurveExt>::AffineExt = b.into();
+
+                    assert!(a == a);
+                    assert!(b == b);
+                    assert!(a != b);
+                    assert!(b != a);
+                }
+            }
         }
-    }
-}
 
-pub fn random_serialization_test<G: CurveExt>()
-where
-    G: SerdeObject,
-    G::AffineExt: SerdeObject,
-{
-    for _ in 0..100 {
-        let projective_point = G::random(OsRng);
-        let affine_point: G::AffineExt = projective_point.into();
-
-        let projective_bytes = projective_point.to_raw_bytes();
-        let projective_point_rec = G::from_raw_bytes(&projective_bytes).unwrap();
-        assert_eq!(projective_point, projective_point_rec);
-        let mut buf = Vec::new();
-        projective_point.write_raw(&mut buf).unwrap();
-        let projective_point_rec = G::read_raw(&mut &buf[..]).unwrap();
-        assert_eq!(projective_point, projective_point_rec);
-
-        let affine_bytes = affine_point.to_raw_bytes();
-        let affine_point_rec = G::AffineExt::from_raw_bytes(&affine_bytes).unwrap();
-        assert_eq!(affine_point, affine_point_rec);
-        let mut buf = Vec::new();
-        affine_point.write_raw(&mut buf).unwrap();
-        let affine_point_rec = G::AffineExt::read_raw(&mut &buf[..]).unwrap();
-        assert_eq!(affine_point, affine_point_rec);
-    }
-}
+        macro_rules! projective_to_affine_affine_to_projective {
```

Comment:
```suggestion
        macro_rules! projective_affine_roundtrip {
```

Is shorter and means the same

Reply:
Done https://github.com/privacy-scaling-explorations/halo2curves/pull/129/commits/71069a710d4b6b6ba268cf5032a177ee7cd02877